Sources/Penny/Services/CoinService/DefaultCoinService.swift

```diff
@@ -69,64 +77,46 @@ actor DefaultCoinService: CoinService {
         
         let body = try await response.body.collect(upTo: 1 << 24)
         
-        return try decoder.decode(Int.self, from: body)
+        return try decoder.decode(CoinResponse.self, from: body)
     }
 
-    func getGitHubID(of user: String) async throws -> GitHubUserResponse {
-        var request = HTTPClientRequest(url: "\(Constants.apiBaseUrl!)/coin")
-        request.method = .POST
-        request.headers.add(name: "Content-Type", value: "application/json")
-        let data = try encoder.encode(CoinRequest.getGitHubID(user: user))
-        request.body = .bytes(data)
-        let response = try await httpClient.execute(request, timeout: .seconds(30), logger: self.logger)
-        logger.trace("Received HTTP response", metadata: ["response": "\(response)"])
+    func getCoinCount(of discordID: UserSnowflake) async throws -> Int {
+        try await self.getUser(discordID: discordID).coinCount
+    }
 
-        guard (200..<300).contains(response.status.code) else {
-            let collected = try? await response.body.collect(upTo: 1 << 16)
-            let body = collected.map { String(buffer: $0) } ?? "nil"
-            logger.error("Get-GitHub-id failed", metadata: [
-                "status": "\(response.status)",
-                "headers": "\(response.headers)",
-                "body": "\(body)",
-            ])
-            throw ServiceError.badStatus(response.status)
+    func getGitHubName(of discordID: UserSnowflake) async throws -> GitHubUserResponse {
+        let user = try await self.getUser(discordID: discordID)
+
+        guard let id = user.githubID else {
+            return .notLinked
         }
 
+        let encodedID = id.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) ?? id
+        let url = "https://api.github.com/user/\(encodedID)"
+        logger.debug("Will make a request to get GitHub user name", metadata: [
+            "user": "\(user)",
+            "url": .string(url),
+        ])
+        var request = HTTPClientRequest(url: url)
+        request.headers = [
+            "Accept": "application/vnd.github+json",
+            "X-GitHub-Api-Version": "2022-11-28",
+            "User-Agent": "Penny/1.0.0 (https://github.com/vapor/penny-bot)"
```

Comment:
I do not at all love that this is hardcoded. (Nor for that matter the GitHub API version.)

Reply:
What to do then?
It's not like it really matters to us.